src/biz/tugay/groovyship/modal/Ship.groovy

```diff
@@ -0,0 +1,20 @@
+package biz.tugay.groovyship.modal
+
+/**
+ * A Ship object is represented as a Map of Coordinate:Boolean where the Boolean value represents whether
+ * the ship coordinate has been hit by a missile or not.
+ * An example might be: [
+ *   [0,0]:false,
+ *   [1,0]:false,
+ *   [2,0]:true
+ * ].
+ * That would be a ship floating horizontally on the first row of a board, where its last piece has been damaged.
+ */
+class Ship
+{
+  Map<Coordinate, Boolean> coordinateIsHitByMissileMap = [:]
+
+  Ship(Coordinate... coordinates) {
+    coordinates.each { coordinateIsHitByMissileMap.put it, false }
```

Comment:
The `{ ... }` syntax is a closure: http://www.groovy-lang.org/closures.html It is another powerful feature of Groovy.

Reply:
yes, i dont know all details but it will be very similar to python